gammapy/irf/energy_dispersion.py

```diff
@@ -370,12 +366,22 @@ def get_bias(self, e_true):
 
     def _get_mean(self, e_true):
         """Get mean log reconstructed energy."""
-        # Reconstructed energy is 1st moment of PDF
         pdf = self.data.evaluate(e_true=e_true)
         norm = np.sum(pdf, axis=pdf.ndim-1)
         temp = np.sum(pdf * self.e_reco._interp_nodes(), axis=pdf.ndim-1)
         return temp / norm
 
+    def _get_variance(self, e_true):
+        """Get variance of log reconstructed energy."""
+        pdf = self.data.evaluate(e_true=e_true)
+        mean = self._get_mean(e_true)
+        erec = self.e_reco._interp_nodes()
+        erec = np.repeat(erec, max(np.sum(mean.shape), 1)).reshape(erec.shape + mean.shape)
```

Comment:
If I would have to moan about something, I'd ask you to put some comments here, what all this is doing. It's kind of annoying (at least for me) to have to decipher numpy expression all the time :smile:

Reply:
Yeah, this is a bit ugly, but I guess this can hardly be avoided if N-dim arrays are supposed to work. The documentation I just added should make things a bit clearer, I hope.